crates/bevy_gltf/src/loader.rs

```diff
@@ -247,36 +247,37 @@ async fn load_gltf<'a, 'b>(
         .collect();
 
     // TODO: use the threaded impl on wasm once wasm thread pool doesn't deadlock on it
-    #[cfg(target_arch = "wasm32")]
-    for gltf_texture in gltf.textures() {
-        let (texture, label) =
-            load_texture(gltf_texture, &buffer_data, &linear_textures, &load_context).await?;
-        load_context.set_labeled_asset(&label, LoadedAsset::new(texture));
-    }
-
-    #[cfg(not(target_arch = "wasm32"))]
-    load_context
-        .task_pool()
-        .scope(|scope| {
-            gltf.textures().for_each(|gltf_texture| {
-                let linear_textures = &linear_textures;
-                let load_context: &LoadContext = load_context;
-                let buffer_data = &buffer_data;
-                scope.spawn(async move {
-                    load_texture(gltf_texture, buffer_data, linear_textures, load_context).await
+    if gltf.textures().len() == 1 || cfg!(target_arch = "wasm32") {
```

Comment:
This could use a quick comment to explain why we want to branch like this.

The fact that we don't get threads on wasm is particularly subtle.

Reply:
I actually opened an issue for that discussion https://github.com/bevyengine/bevy/issues/1924, linking it here 🙂